### PR TITLE
enable debug builds for ORT

### PR DESCRIPTION
**Pull request by askhade**
Adding an option to build ort in debug mode to easily help with debugging. Intentionally added a separate option for ORT "TRITON_BUILD_ONNXRUNTIME_CONFIG" as opposed to just using CMAKE_BUILD_TYPE to enable building ort-backend bridge and ort backend in different modes.

### CODE REVIEW

File: tools/gen_ort_dockerfile.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def dockerfile_common():
 ARG BASE_IMAGE={}
 ARG ONNXRUNTIME_VERSION={}
 ARG ONNXRUNTIME_REPO=https://github.com/microsoft/onnxruntime
-'''.format(FLAGS.triton_container, FLAGS.ort_version)
+ARG ONNXRUNTIME_BUILD_TYPE={}
```
Comment (deadeyegoodwin) [inline tools/gen_ort_dockerfile.py:47]: Be consistent with name, use ONNXRUNTIME_BUILD_CONFIG

File: tools/gen_ort_dockerfile.py [hunk 2]
```diff
@@ -292,7 +298,7 @@ def dockerfile_for_windows(output_file):
 WORKDIR /workspace/onnxruntime
 ARG VS_DEVCMD_BAT="\BuildTools\Common7\Tools\VsDevCmd.bat"
 RUN powershell Set-Content 'build.bat' -value 'call %VS_DEVCMD_BAT%',(Get-Content 'build.bat')
-RUN build.bat --cmake_generator "Visual Studio 16 2019" --config Release --skip_submodule_sync --build_shared_lib --use_openmp --update --build --build_dir /workspace/build {}
+RUN build.bat --cmake_generator "Visual Studio 16 2019" --config ${ONNXRUNTIME_BUILD_TYPE} --skip_submodule_sync --build_shared_lib --use_openmp --update --build --build_dir /workspace/build {}
```
Comment (deadeyegoodwin) [inline tools/gen_ort_dockerfile.py:301]: In windows docker you have to use %varname% to access variables.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -76,6 +76,7 @@ option(TRITON_ENABLE_ONNXRUNTIME_OPENVINO
 set(TRITON_BUILD_CONTAINER "" CACHE STRING "Triton container to use a base for build")
 set(TRITON_BUILD_CONTAINER_VERSION "" CACHE STRING "Triton container version to target")
 set(TRITON_BUILD_ONNXRUNTIME_VERSION "" CACHE STRING "ONNXRuntime version to build")
+set(TRITON_BUILD_ONNXRUNTIME_CONFIG "" CACHE STRING "ONNXRuntime config to build. Debug, Release, RelWithDebInfo")
```
Comment (deadeyegoodwin) [inline CMakeLists.txt:79]: Can you just use CMAKE_BUILD_TYPE instead of introducing a new variable.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (askhade) [reply CMakeLists.txt:79]: My initial intention to keep this separate was to enable building the bridge and backend in different modes. Rethinking this such an ability is not necessary... Will merge it.

**Review by deadeyegoodwin — APPROVED** [review]

**Comment by deadeyegoodwin** [comment]
Can you confirm that backend builds correctly with this change and then I will merge.

**Comment by askhade** [comment]
I have tested the builds on Linux but not on windows... Let me try building on windows today... Thanks!

**Comment by deadeyegoodwin** [comment]
@askhade  Did this build ok on windows.

**Comment by askhade** [comment]
I was running into different build issues (unrelated to this PR as I am building on windows for the first time) ... I am planning to finish this test tomorrow... will update this PR as soon as I am done... Thanks!

**Comment by askhade** [comment]
PR #80 enables debug builds for Linux. Windows work will be done later.

**Pull request closed without merging**